### ISSUE TITLE
Large organization support

### ISSUE THREAD

**Issue by paulslaby** [derberg/manage-files-in-multiple-repositories#66]
I have a problem. I want to distribute a small portion of files (currently one) into a large number of repositories. I am hitting the limit 10GB of disk space on the worker. 
Possible solutions I see: 
- no cloning of the target repo (we can create a PR directly via API)
- deleting the cloned repo after processing is done (quickwin, but breaks caching possibility)
- splitting repositories into batches and run batches in distinct runs

**Comment by derberg** [comment]
> 10GB of disk space

😅 

> we can create a PR directly via API

didn't know that is possible. Not sure how that would work though

> deleting the cloned repo after processing is done (quickwin, but breaks caching possibility)

I think this one is easiest and caching is probably not a big deal

**Comment by paulslaby** [comment]
I rewrote it in a fork in our company: https://github.com/customink/manage-files-in-multiple-repositories
Feel free to get inspired, or anything you want. But i think those changes are too big to create a PR.

**Comment by Ryangr0** [comment]
I would really like to do the same. Our situation is as follows: we have many (large) repositories. Cloning them sometimes fails, for different reasons. (Network issues, crashing runner pods, etc) That means that the changes in our workflows don't get applied and there's no real visibility on this. I'm thinking about many different ways to solve this problem such as retry mechanisms, alerting, etc, but I'd also like to inquire about whether your fork is still "active" and if it's reliable enough for usage, before I spend a lot of time testing it @paulslaby ? As I see that it has been archived. That would really help my specific use case!

**Comment by paulslaby** [comment]
TLDR: We don't use it anymore. But we used to with no issues. 

Currently we paid for a application service catalog SaaS and we needed to create a very small service to check repository contents. This script already had most of the features needed for distributing a file across repositories in the whole org.
So don't expect any progress on that script development, but should be usable as is. Of course network issues might disturb it, but that job would be red and re-run should fix it (idempotent). 
And archival is because we don't want to maintain it, when we don't use it anymore.
